### PR TITLE
Refactor constants into modules

### PR DESCRIPTION
**Pull request by rvyuha**
Refactored contants.R into 3 new module files.
- dictionary-module.R Stores strings for dictionary sheets access as well as different dictionary-specific qmds.
- parts-module.R Stores strings related to the dictionary parts sheet.
- sets-module.R Stores string related to the dictionary sets sheet.
- Removed references to constants.R with new module files
- Removed constants.R

### CODE REVIEW

**Review by yulric — CHANGES_REQUESTED** [review]
* Can we standardize how we represent the information for the columns in a sheet? This can be done in another PR
	* I propose the following schema:
		* ```
		  list(
			  <column_list_name> = list(
                    // Contains the name of the column in the sheet
                    name = <column_name>,
				  // This is added only if the column is categorical
				  categories = list(
					  <category_list_name> = <category_identifier>
				  )
			  )
		  )
		* For the parts sheet the `list` would be (this list is abbreviated),
			* ```
			  list(
				  part_id = list(
					  name = "partID",
				  ),
				  part_type = list(
					  name = "partType",
					  categories = list(
						  tables = "tables"
					  )
				  )
			  )
		* Dynamic columns, columns whose names we don't know beforehand, like the table columns in the parts sheet are problematic.
			* We can store the information for those columns in a variable and then dynamically add to the schema in the code
			* For example, 
			```
			 table_columns_metadata <- list(
			     categories = list(
				primary_key = "pK",
				foreign_key = "fK",
				header = "header",
				not_applicable = "N/A"
			     )
		     )
	* We would do the same for the sets sheet
* Can we remove the `module` string from the file names. The fact that they are a file already identifies them as a module.
	* parts-module -> parts-sheet
	* sets-module -> sets-sheet
	* dictionary-module -> odm-dictionary-file

**Comment by rvyuha** [comment]
> * Can we standardize how we represent the information for the columns in a sheet? This can be done in another PR
>   
>   * I propose the following schema:
>     
>     * ```
>       list(
>           <column_list_name> = list(
>                 // Contains the name of the column in the sheet
>                 name = <column_name>,
>         	  // This is added only if the column is categorical
>         	  categories = list(
>         		  <category_list_name> = <category_identifier>
>         	  )
>           )
>       )
>       ```
>     * For the parts sheet the `list` would be (this list is abbreviated),
>       
>       * ```
>         list(
>             part_id = list(
>           	  name = "partID",
>             ),
>             part_type = list(
>           	  name = "partType",
>           	  categories = list(
>           		  tables = "tables"
>           	  )
>             )
>         )
>         ```
>     * Dynamic columns, columns whose names we don't know beforehand, like the table columns in the parts sheet are problematic.
>       
>       * We can store the information for those columns in a variable and then dynamically add to the schema in the code
>       * For example,
>       
>       ```
>         table_columns_metadata <- list(
>             categories = list(
>        	primary_key = "pK",
>        	foreign_key = "fK",
>        	header = "header",
>        	not_applicable = "N/A"
>             )
>         )
>       ```
>   * We would do the same for the sets sheet
> * Can we remove the `module` string from the file names. The fact that they are a file already identifies them as a module.
>   
>   * parts-module -> parts-sheet
>   * sets-module -> sets-sheet
>   * dictionary-module -> odm-dictionary-file

I agree with the schema approach however I am not sure if the dynamic columns even need to be stored?
I also agree that module is not the best suffix for the file name, but I still believe it would be best to distinguish that this file only stores strings related to that sheet not any functions.

**Comment by rvyuha** [comment]
I noticed that most of your comments are about moving strings into function related files? I am not sure that is the best approach as it would mix string storage and functions. Is this the best approach for scalability? Wouldn't it make finding strings to modify more difficult?

**Comment by yulric** [comment]
> I noticed that most of your comments are about moving strings into function related files? I am not sure that is the best approach as it would mix string storage and functions. Is this the best approach for scalability? Wouldn't it make finding strings to modify more difficult?

Deciding where to put a constant/function depends on the domain it deals with. For example the constants for parts sheet and functions that deal with the parts sheet should be in one module.

File: R/dictionary-module.R [hunk 1]
```diff
@@ -0,0 +1,8 @@
+dictionary <- list()
+
+dictionary$dictionary_directory <- "data/raw"
```
[the remark this answers is not in the document]
Comment (rvyuha) [reply R/dictionary-module.R:3]: Will move it once the merge occurs.

File: qmd/tables.qmd [hunk 2]
```diff
@@ -63,15 +63,18 @@ source("R/qmd-utils.R")
 source("R/format-table.R")
 # Source warning utils
 source("R/warning-utils.R")
-source("R/dictionary-module.R")
-source("R/parts-module.R")
-
+source("R/odm-dictionary-file.R")
+source("R/parts-sheet.R")
+# String set Values
+parts$part_sheet_table_column_type_set_PK <- "Primary Key"
```
Comment (yulric) [inline qmd/tables.qmd:69]: Why are you adding these to the `parts` object rather than as independent variables?

File: R/qmd-utils.R [hunk 3]
```diff
@@ -133,8 +133,10 @@ create_html_list <- function(string_to_append, list_values){
 #' 
 #' @return string with sets or parts depending on suffix of input_ID
 determine_link_source <- function(input_ID){
-  link_to_parts <- dictionary$parts_qmd_file_name
-  link_to_sets <- dictionary$sets_qmd_file_name
+  odm_dictionary$parts_qmd_file_name <- "parts"
```
Comment (yulric) [inline R/qmd-utils.R:136]: Why are you add these to the `odm_dictionary` object? Why can't you store them as independent variables?

**Review by yulric — CHANGES_REQUESTED** [review]

**Comment by rvyuha** [comment]
@yulric Ready for review

**Review by yulric — APPROVED** [review]

**Pull request merged**